pkg/cmd/taskrun/logs_test.go

```diff
@@ -0,0 +1,150 @@
+// Copyright © 2019 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package taskrun
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/jonboulle/clockwork"
+	"github.com/knative/pkg/apis"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/logs"
+	tu "github.com/tektoncd/cli/pkg/test"
+	cb "github.com/tektoncd/cli/pkg/test/builder"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/test"
+	tb "github.com/tektoncd/pipeline/test/builder"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestLog_no_taskrun_arg(t *testing.T) {
+	c := Command(&tu.Params{})
+
+	_, err := tu.ExecuteCommand(c, "logs", "-n", "ns")
+	if err == nil {
+		t.Error("Expecting an error but it's empty")
+	}
+}
+
+func TestLog_missing_taskrun(t *testing.T) {
+	tr := []*v1alpha1.TaskRun{
+		tb.TaskRun("output-taskrun-1", "ns"),
+	}
+	cs, _ := test.SeedTestData(test.Data{TaskRuns: tr})
+	p := &tu.Params{Tekton: cs.Pipeline, Kube: cs.Kube}
+
+	c := Command(p)
+	got, _ := tu.ExecuteCommand(c, "logs", "output-taskrun-2", "-n", "ns")
+	expected := msgTRNotFoundErr + " : taskruns.tekton.dev \"output-taskrun-2\" not found \n"
+
+	if d := cmp.Diff(expected, got); d != "" {
+		t.Errorf("Unexpected output mismatch: \n%s\n", d)
+	}
+}
+
+func TestLog_valid_taskrun_logs(t *testing.T) {
+	var (
+		prstart      = clockwork.NewFakeClock()
```

Comment:
could drop `prstart` altogether?

Reply:
This is required to set Start Time for Taskrun.

Reply:
can use directly [here](https://github.com/tektoncd/cli/pull/52/files#diff-ef455c63ca0692aba6213e033f2e8379R67)?
`clockwork.NewFakeClock().Add(20 * time.Second)` ?

Reply:
Done